encord/client_metadata_schema.py

```diff
@@ -1,26 +1,83 @@
-from typing import Dict, Optional
+import functools
+from typing import Any, Dict, Optional
 
-import encord.orm.client_metadata_schema as orm
+from encord.exceptions import MetadataValidationError
 from encord.http.v2.api_client import ApiClient
+from encord.orm.client_metadata_schema import (
+    ClientMetadataSchema,
+    ClientMetadataSchemaPayload,
+    ClientMetadataSchemaTypes,
+    validate_embedding,
+)
 
 
-def set_client_metadata_schema_from_dict(api_client: ApiClient, json_dict: Dict[str, orm.ClientMetadataSchemaTypes]):
+@functools.lru_cache(maxsize=128)
+def cached_get_client_metadata_schema(api_client: ApiClient) -> Optional[Dict[str, ClientMetadataSchemaTypes]]:
+    return get_client_metadata_schema(api_client=api_client)
+
+
+def _metadata_validation_error_message_builder(key: str, schema_type: ClientMetadataSchemaTypes, value: Any):
+    message = f"Failed to validate at {key=} against {schema_type=}, where we recieved {value=}"
+    return message
+
+
+def validate_metadata(
```

Comment:
Is there any point having the function return a bool when in actuality it'll never return False. As is, it'll either return True or it'll throw an MetadataValiationError. Interested in thoughts